operator/src/main/java/com/instaclustr/cassandra/operator/jmx/CassandraConnectionFactory.java

```diff
@@ -0,0 +1,83 @@
+package com.instaclustr.cassandra.operator.jmx;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.cache.RemovalNotification;
+import com.google.common.net.InetAddresses;
+import io.kubernetes.client.models.V1Pod;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.management.remote.JMXConnectionNotification;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+public class CassandraConnectionFactory {
+    private static final Logger logger = LoggerFactory.getLogger(CassandraConnectionFactory.class);
+
+    private final LoadingCache<InetAddress, CassandraConnection> connectionCache = CacheBuilder.<InetAddress, CassandraConnection>newBuilder()
+            .expireAfterWrite(10, TimeUnit.MINUTES)
+            .removalListener((RemovalNotification<InetAddress, CassandraConnection> notification) -> {
+                try {
+                    notification.getValue().close();
+
+                } catch (final IOException e) {
+                    logger.warn("Failed to close JMX connection to {}.", notification.getKey(), e);
+                }
+            })
+            .build(new CacheLoader<InetAddress, CassandraConnection>() {
+                @Override
+                public CassandraConnection load(@Nonnull final InetAddress key) throws Exception {
+                    final JMXServiceURL serviceURL = new JMXServiceURL(String.format("service:jmx:rmi:///jndi/rmi://%s:7199/jmxrmi", InetAddresses.toAddrString(key)));
```

Comment:
port is hard coded, let's make that configurable if possible

Reply:
It _probably_ should be discovered from the Service, if possible.

Reply:
ok let's address when we make jmx configurable either via configMaps or the CRD (that way its set at the service on creation).

LGTM